New board support

RailsRoyale opened the pull request:
Updating board list and descriptions with new assets to support new 4chan boards.

RailsRoyale committed:
New board additions

Updating board list with new assets to support new 4chan boards.

snoozbuster commented:
Hey, how did you get the project to build? I had an unspecified error when trying to build vpx_lib.

RailsRoyale commented:
The project didn't work for me out of the box so there was a bunch of things I had to adjust on my side. IIRC my vpx_lib issues went away when I downloaded VS2012 because of some missing dll.

Do you have any other output? Trying to jog my memory.

snoozbuster commented:
I have VS2015 Community. When I try to build vpx_arm I get to the step "Assembling vp9_reconintra_neon_asm.asm.s" and then get "error MSB6006: "cmd.exe" exited with code -1073741515". Trying to build vpx_win32 I get a lot of messages about the yasm tool being missing (those messages aren't present when trying to build the ARM version). Got any ideas? 

Also, where'd you get those board images? They're perfect.

RailsRoyale commented:
Ah, I see. Yeah, the cmd.exe exit code -1073741515 was caused by a missing dll called msvcdis110.dll (%PROGRAMFILES%\Microsoft Visual Studio 11.0\Common7\IDE\msvcdis110.dll), but that error is obfuscated unless you try building it via the developer console. You could try downloading the VS2012 Redistribution Package or VS2012 Express to get that dll.

I never got the vpx_win32 variant building so I only tested it on the actual phone. I tried downloading yasm and cygwin and pointing at those tools, but it wouldn't take. Would have to see how pantaloons set up his build most likely.

As for the pictures, I pulled /trash/ and /aco/ from the board's sticky. The rest were images I searched for that were inspired by the general topics on the board at the time.

snoozbuster commented:
Interesting. Thanks for that! I'll see if I can get it building, I would love to fix the captcha problem. Hope pantaloons takes notice and updates the app on the store. At the very least I can have an updated version!

snoozbuster commented:
Just in case you're curious, the VS2012 redist didn't work, I had to install VS2012 Express. However now the nestegg library is complaining about requiring a retarget to WP8.1, did you have to do that? It doesn't bother to explain why that would be required.

snoozbuster commented:
Also, when the app runs it fails to load any threads because it can't find System.Xml.Xpath. I tried removing the reference to Xpath and it doesn't seem like it's used anywhere in the code but regardless of how many times I clean and rebuild it still crashes when entering `Post.Merge(APIPost)` from `Post.cs:206`. Did you encounter this?

RailsRoyale commented:
It prompts for a retarget when you open the solution for the first time, so if you accidentally said yes that could be part of it. 

As for the Xml path issue, I never encountered it myself. You may need to download a certain .NET or Phone SDK version to reference it. I can't say for sure without a more specified error.

I wouldn't remove a reference for the sake of consistency with either future syncs or potential pull requests.

-----Original Message-----
From: "Alex Van Liew" notifications@github.com
Sent: ‎1/‎11/‎2016 6:23 PM
To: "pantaloons/4charm" 4charm@noreply.github.com
Cc: "RailsRoyale" albertvilam@gmail.com
Subject: Re: [4charm] New board support (#14)

Also, when the app runs it fails to load any threads because it can't find System.Xml.Xpath. I tried removing the reference to Xpath and it doesn't seem like it's used anywhere in the code but regardless of how many times I clean and rebuild it still crashes when entering Post.Merge(APIPost) from Post.cs:206. Did you encounter this?
—
Reply to this email directly or view it on GitHub.

snoozbuster commented:
All the literature I've found online indicates that System.Xml.Xpath simply can't be used in Windows Store or Windows Phone apps, . I see the package in NuGet but it won't install because it's not targeting the full .NET framework. My copy of VS2015 community just says "referenced component cannot be found" but I think HTMLAgilityPack depends on it (which is why it's strange that that installation went fine). What version of VS are you using?

RailsRoyale commented:
I'm using VS2013 Ultimate, but I don't think that's the root of it.

I did have a problem with NuGet already existing in the directory (.nuget) which is an old version which caused my package manager to throw a fit. I deleted that folder and updated the NuGet manager in VS itself which then fixed all of my dependent packages like HTMLAgilityPack. 

If you have the pre-existing .nuget folder you could remove it from the directory temporarily and see if NuGet allows some sort of update that could potentially fix your problem.

snoozbuster commented:
VS2015 didn't complain about NuGet at all, weird that yours did. Deleting the folder didn't help but I found a package by Microsoft `System.Xml.XPath.SL4` that appears to add the appropriate assembly (it's a copy of an assembly found in the Silverlight 4 SDK), and HTMLAgilityPack mentions that a reference to that particular assembly is required for WP in its readme. That's pretty much all I needed (other than the file from VS2012), now it works. Thanks for your help. I'm going to make a pull request with a few minor changes I made to make it build, would you try them on your machine to see if it still works on earlier versions of VS?

RailsRoyale commented:
Go ahead. I'll give it a shot in the next few days and see if it builds when I have time.

Ideally pantaloons will get on at some point and we can update the solutions and targets to 8.1 and perhaps a Win10 variant.

I hope he merges this change at some point because I have a potential solution for the captcha problem but I would like his opinion to see if it's appropriate.

-----Original Message-----
From: "Alex Van Liew" notifications@github.com
Sent: ‎1/‎12/‎2016 1:53 AM
To: "pantaloons/4charm" 4charm@noreply.github.com
Cc: "RailsRoyale" albertvilam@gmail.com
Subject: Re: [4charm] New board support (#14)

VS2015 didn't complain about NuGet at all, weird that yours did. Deleting the folder didn't help but I found a package by Microsoft System.Xml.XPath.SL4 that appears to add the appropriate assembly (it's a copy of an assembly found in the Silverlight 4 SDK), and HTMLAgilityPack mentions that a reference to that particular assembly is required for WP in its readme. That's pretty much all I needed (other than the file from VS2012), now it works. Thanks for your help. I'm going to make a pull request with a few minor changes I made to make it build, would you try them on your machine to see if it still works on earlier versions of VS?
—
Reply to this email directly or view it on GitHub.

snoozbuster commented:
I took a peek at updating to 8.1 and it's a bit more involved than one would hope. The "Retarget to Windows Phone 8.1" option retargets it as a Silverlight app, and as luck would have it you can't use System.Net.Http in a Silverlight app, only in a Runtime app. I wasn't able to find any docs about how to actually upgrade to a Runtime app, though, just differences. Upgrading to Silverlight would mean rewriting a good chunk of the HttpClient code. 

I have my own thoughts about a captcha fix. We should probably discuss them in #12, though. I'll comment my thoughts and findings there once I write them up.
